subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -687,6 +690,30 @@ public List<SubscriptionBillingEvent> getSubscriptionBillingEvents(final Version
         }
     }
 
+
+    private DateTime alignToNextBCD(final Plan curPlan, final PlanPhase curPlanPhase, final DateTime originalDate, final SubscriptionCatalog catalog, final Integer bcdLocal, final InternalTenantContext context) throws SubscriptionBaseApiException, CatalogApiException {
+
+        final BillingAlignment billingAlignment = catalog.billingAlignment(new PlanPhaseSpecifier(curPlan.getName(), curPlanPhase.getPhaseType()),
```

Comment:
This is not new (I think it came up when we were doing https://github.com/killbill/killbill/issues/784), but it's not clear whether we want the `prev` or `next` plan here (e.g. after change plan) to find the billing alignment.

Reply:
I think in this case, the problem does not exist as we are not going from a `prev` -> to a `next` (this would happen if we were to look for 'change of plan' alignment rules).  In this case, given the `cur` Plan/Phase return the billing alignment.